src/lib/datepicker/datepicker.ts

```diff
@@ -229,9 +235,9 @@ export class MatDatepicker<D> implements OnDestroy, CanColor {
   id: string = `mat-datepicker-${datepickerUid++}`;
 
   /** The currently selected date. */
-  get _selected(): D | null { return this._validSelected; }
-  set _selected(value: D | null) { this._validSelected = value; }
-  private _validSelected: D | null = null;
+  get _selected(): MatDateSelection<D> { return this._validSelected; }
```

Comment:
Should we change the type back here? I don't think we want anything changing out the selection model